### PR TITLE
Swift: Add integral type classes

### PR DESCRIPTION
**Pull request by MathiasVP**
I had these lying around as left-overs from some unrelated work, and they seem too useful to throw away (since most other languages have versions of these).

### CODE REVIEW

**Review by geoffw0 — COMMENTED** [review]
Thanks for finding and sharing this work.  I think it needs a bit more polish before we merge it though, e.g. there really ought to be minimal test coverage.  I know you have other priorities - let me know if you'd prefer me to do the changes to this PR?

File: swift/ql/lib/swift.qll [hunk 1]
```diff
@@ -10,4 +10,5 @@ import codeql.swift.elements.expr.InitializerCallExpr
 import codeql.swift.elements.expr.SelfRefExpr
 import codeql.swift.elements.decl.MethodDecl
 import codeql.swift.elements.decl.ClassOrStructDecl
+import codeql.swift.elements.type.NumericOrCharType
```
[the remark this answers is not in the document]
Comment (MathiasVP) [reply swift/ql/lib/swift.qll:13]: Doh. Thanks!

File: swift/ql/lib/codeql/swift/elements/type/NumericOrCharType.qll [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+/** The `Bool` type. */
+class BooleanType extends Type {
+  BooleanType() { this.getName() = "Bool" }
+}
+
+/**
+ * A numeric-like type. This includes the types `Character`, `Bool`, and all
+ * the integer-like types.
```
[the remark this answers is not in the document]
Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/elements/type/NumericOrCharType.qll:27]: Good point. I agree with both of these observations 👍

**Comment by MathiasVP** [comment]
> Thanks for finding and sharing this work. I think it needs a bit more polish before we merge it though, e.g. there really ought to be minimal test coverage. I know you have other priorities - let me know if you'd prefer me to do the changes to this PR?

Thanks! Yeah, I agree with all those comments. I'm not sure what the best testing approach to this would be. A naive thing would be a ql test that just selects all the relevant types from a large `.swift` file, but I fear that this will just be a `.expected` file that we just blindly accept test changes to.

I'd really appreciate if you could take over this PR 🙇.

**Comment by geoffw0** [comment]
No problem, I'll finish this PR probably this week.

File: swift/ql/lib/codeql/swift/elements/type/NumericOrCharType.qll [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  CharacterType() { this.getName() = "Character" }
+}
+
+/**
+ * An integer-like type. For example, `Int`, `Int16`, `Uint16`, etc.
+ */
+class IntegerType extends Type {
+  IntegerType() {
+    this.getName() =
+      ["Int", "Int8", "Int16", "Int32", "Int64", "UInt", "UInt8", "Uint16", "Uint32", "UInt64"]
```
Comment (atorralba) [inline swift/ql/lib/codeql/swift/elements/type/NumericOrCharType.qll:14]: These seem like typos.

```suggestion
      ["Int", "Int8", "Int16", "Int32", "Int64", "UInt", "UInt8", "UInt16", "UInt32", "UInt64"]
```

Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/elements/type/NumericOrCharType.qll:14]: Oh, yes. Well spotted!

**Comment by geoffw0** [comment]
I've made a PR onto this PR here: https://github.com/MathiasVP/ql/pull/5

@MathiasVP I suggest you review that and merge it (into this PR) when you're happy, then I'll merge this PR into main, hence everything will have had 2+ sets of eyes on it.

Any other reviewers, you're welcome to comment on there or here as well.

**Review by geoffw0 — APPROVED** [review]
:+1:

I will do a follow-up to use the new classes in queries.

**Comment by geoffw0** [comment]
Checks failed - "Unable to connect to azure.archive.ubuntu.com" - sounds like we should retry in a few hours?

**Comment by MathiasVP** [comment]
> Checks failed - "Unable to connect to azure.archive.ubuntu.com" - sounds like we should retry in a few hours?

Yeah. I already retried some other failing CI checks, and they seemed to succeed now. I'll just try restarting the final one right away.

**Comment by MathiasVP** [comment]
Bahh... Failed again 😭.

**Pull request merged**